cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -266,15 +273,7 @@ float DrawableSubmap::UpdateAlpha(const float target_alpha) {
 }
 
 void DrawableSubmap::ToggleVisibility() {
-  if (visibility_->getBool()) {
-    if (scene_node_->numAttachedObjects() == 0) {
-      scene_node_->attachObject(manual_object_);
-    }
-  } else {
-    if (scene_node_->numAttachedObjects() > 0) {
-      scene_node_->detachObject(manual_object_);
-    }
-  }
+  scene_node_->setVisible(visibility_->getBool());
```

Comment:
Nah, I just didn't know about `setVisible()`. Plus, now I can handle visibility of both axes and the submap by having them both under a signle `scene_node_`.